src/orchestrator.ts

```diff
@@ -601,58 +611,68 @@ export class Orchestrator {
     private findSubOrchestrationInstanceCreated(
         state: HistoryEvent[],
         name: string,
-        instanceId: string)
+        instanceId: string | undefined)
         : SubOrchestrationInstanceCreatedEvent | undefined {
+        const matches = state.filter((val: HistoryEvent) => {
+            return val.EventType === HistoryEventType.SubOrchestrationInstanceCreated
+                && !val.IsProcessed;
```

Comment:
So the basic idea of this iteration is that you don't need to compare names since you can simply rely on the order of these events, right?

Reply:
That is the core idea.

I do some extra validation to error out when I detect that the order seems to be incorrect based on the fact that the history doesn't match what we are trying to execute.